AC-817 Create FormEntity and update other entity classes

rishabh-997 opened the pull request:
<!--- Add a pull request title above in this format -->
<!--- real example: 'AC-204 Applying MVP Model' -->
<!--- 'AC-JiraIssueNumber JiraIssueTitle' -->

## Description of what I changed
1. Added formEntity to replace ActiveAndroid.
2. Looked through entire code to create queries in FormDAO interface
3. Updated Patient and Encounter Entities
4. Added new and updated previous tests depending on entity
<!--- Describe your changes in detail -->
<!--- It can simply be your commit message, which you must have -->

## Issue I worked on
<!--- This project only accepts pull requests related to open issues -->
<!--- Want a new feature or change? Discuss it in an issue first -->
<!--- Found a bug? Point us to the issue/or create one so we can reproduce it -->
<!--- Just add the issue number at the end: -->
JIRA Issue: https://issues.openmrs.org/browse/AC-817

## Checklist: I completed these to help reviewers :)
<!--- Put an `x` in the box if you did the task -->
<!--- If you forgot a task please follow the instructions below -->
- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).
<!--- No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one -->

- [x] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)
<!--- No? -> write tests and add them to this commit `git add . && git commit --amend`-->

- [x] All new and existing **tests passed**.
<!--- No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works. -->

- [x] My pull request is **based on the latest changes** of the master branch.
<!--- No? Unsure? -> execute command `git pull --rebase upstream master` -->

LuGO0 commented on openmrs-client/src/main/java/org/openmrs/mobile/dao/FormResourceDAO.java:
```diff
@@ -0,0 +1,43 @@
+
+    @Insert
+    void saveFormResource(FormResourceEntity formResourceEntity);
+}
```
addFormResource will be more apt !

rishabh-997 replied:
Actually all other DAO-interfaces uses `saveTable`  naming, so I tried to make similar type of functions have similar naming. I will need to change all the DAO-interfaces with the same in that case

LuGO0 replied:
ohkk @rishabh-997 :+1

f4ww4z replied:
> Actually all other DAO-interfaces uses `saveTable` naming, so I tried to make similar type of functions have similar naming. I will need to change all the DAO-interfaces with the same in that case

@rishabh-997 This is better, can you do this right in this PR?

rishabh-997 replied:
yes. I have updated all the `saveEntity` to `addEntity`

LuGO0 commented on openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/PatientEntity.java:
```diff
@@ -45,6 +45,8 @@
     private String postalCode;
+    @ColumnInfo(name = "dead")
+    private String deceased;
     @Embedded(prefix = "patient_")
```
@rishabh-997 maybe this column name and param name can be made same and something which is more descriptive than simply dead??

rishabh-997 replied:
the column name is the same as that in the server, so I preferred not to change it.

LuGO0 replied:
yes @rishabh-997 I was asking if we can we make the param same to it??

f4ww4z replied:
@rishabh-997 @LuGO0 I think we can leave it as it is for now, we can change it later.

LuGO0 commented on openmrs-client/src/androidTest/java/org/openmrs/mobile/listeners/watcher/dao/PatientRoomDAOTest.java:
```diff
@@ -161,7 +163,7 @@ private PatientEntity newPatientEntity(long id, String uuid, String display,
                                            String identifier, String middleName, String photo,
-                                           String postalCode, String state, boolean synced) {
+                                           String postalCode, String state, boolean synced, String dead) {
         PatientEntity entity = new PatientEntity();
```
if you update that in the entity update it here as well

f4ww4z commented on openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/EncounterEntity.java:
```diff
@@ -19,6 +19,10 @@
     private String formUuid;
+    @ColumnInfo(name = "location_uuid")
+    private String locationUuid;
+    @ColumnInfo(name = "encounterProviders_uuid")
```
```suggestion
    @ColumnInfo(name = "encounter_provider_uuid")
```

rishabh-997 commented:
I have updated PR with requested changes and now merging this PR. I have updated all instances of LocationDAO to LocationRoomDAO and I **need the changes in this PR  to proceed further** with other dao's. I do not have admin rights on JIRA so please close the issue there. If you need me to address more changes here, please add the suggestion and I will incorporate them in the final migration @f4ww4z si :)
